thunder/dynamo/report.py

```diff
@@ -464,3 +471,194 @@ def helper_backend(gm, example_inputs):
     compiled(*args, **kwargs)
 
     return FXReport(graphs)
+
+
+class ThunderSegmentGraphReport(FXGraphReport):
+    def __init__(self, graph, graph_name, compiled_fn, example_input, thunder_options: dict, split_reason: str):
+        super().__init__(graph, graph_name)
+        self.compiled_fn = compiled_fn
+        self.example_input = example_input
+        self.thunder_options = thunder_options
+        self.split_reason = split_reason
+
+        self.fusion_reports: list[ThunderFusionReport] = []
+        self.fwd_trc = None
+        self.bwd_trc = None
+
+    def create_thunder_traces(self):
+        from thunder import last_traces, last_backward_traces
+
+        example_inputs = eval(f"[\n{chr(10).join(arg_like(a) for a in self.example_input)}]")
+        # executes to get the trace
+        run_backward(self.compiled_fn, *example_inputs)
+        self.fwd_trc = last_traces(self.compiled_fn)[-1]
+        self.bwd_trc = last_backward_traces(self.compiled_fn)[-1]
+
+    def create_fusion_reports(self):
```

Comment:
It would nice to have a docstring for this. I couldn't infer from the name that this will populate `fusion_reports` attribute.